x-pack/legacy/plugins/actions/server/actions_config.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fromNullable } from 'fp-ts/lib/Option';
+import { URL } from 'url';
+
+export interface ActionsKibanaConfig {
+  enabled: boolean;
+  whitelistedHosts: 'any' | 'none' | string[];
+}
+
+export interface ActionsConfigurationUtilities {
+  isWhitelistedHostname: (uri: string) => boolean;
+}
+
+export function getActionsConfigurationUtilities(
+  config: ActionsKibanaConfig
+): ActionsConfigurationUtilities {
+  return {
+    isWhitelistedHostname(uri: string): boolean {
+      switch (config.whitelistedHosts) {
+        case 'none':
+          return false;
+        case 'any':
+          return true;
+        default:
+          if (Array.isArray(config.whitelistedHosts)) {
+            const urlHostname = new URL(uri).hostname;
```

Comment:
in some cases like the email service, the value we have is a hostname, which won't work with `new URL()`.  I suppose in those cases we could force the service to pre-pend a bogus `http://` scheme or something.  Or we could somehow make this function allow either a url or hostname, or have two separate functions.

Reply:
Yeah, I'm leaving that to when we change the email task to use this mechanism.
I try to keep the code in the PR relevant to the feature at hand... I'd rather make that change in a subsequent PR.

Reply:
I'm fine handling the other actions whitelisting bits in a subsequent PR, but wondering what shape a whitelist-checking utility function looks like that takes just a hostname?  Separate function, peer of `isWhitelistedHostname()`?  If so, I'd guess I'd want the APIs named `isWhitelistedHostname()` that checks hostnames and `isWhitelistedUrl()` that took URLs, meaning you'd be making a lot of name changes in the PR.  :-)

Reply:
My instinct is actually for it to be done within the same function - any reason to split them apart?

Reply:
how can a single function distinguish whether the input string is a URL (and so needs to be parsed with `URL`) or a hostname?  Or it could be `URL|string` (ie, a URL object)? Try to parse the string as a URL and if it throws or hostname is blank, treat it as a hostname?  I guess I was thinking either two functions, or a single function that takes a "parseAsURL" kind of boolean flag argument, or such.

Reply:
Yeah, I was thinking of using the type ```URL|string``` but if we're treating APIs as JS rather than TS, then probably better to split this to two functions.
I definitely prefer that to a boolean flag.

Reply:
Split into two separate functions